server/src/main/java/com/testsigma/controller/api/agent/AgentDevicesController.java

```diff
@@ -123,23 +129,78 @@ public IosDeveloperImageDTO developer(@PathVariable("agentUuid") String agentUui
     return iosDeveloperImageDTO;
   }
 
-  @RequestMapping(value = "/{deviceUuid}/wda", method = RequestMethod.GET)
+  @RequestMapping(value = "/{deviceUuid}/wda_real_device", method = RequestMethod.GET)
   public IosWdaResponseDTO deviceWdaUrl(@PathVariable String agentUuid, @PathVariable String deviceUuid)
-    throws TestsigmaException {
+          throws TestsigmaException, MalformedURLException {
     log.info(String.format("Received a GET request api/agents/%s/devices/%s/wda", agentUuid, deviceUuid));
     IosWdaResponseDTO iosWdaResponseDTO = new IosWdaResponseDTO();
+
+    ArrayList<Header> headers = new ArrayList<>();
+    headers.add(new BasicHeader(HttpHeaders.CONTENT_TYPE, "application/json"));
+    HttpResponse<String> response = httpClient.get(testsigmaOSConfigService.getUrl() +
+            URLConstants.TESTSIGMA_OS_PUBLIC_WDA_REAL_DEVICE_URL, headers, new TypeReference<>() {
+    });
+    URL wdaRealDeviceURL = new URL(response.getResponseEntity());
+    log.info("Received wda real device remote url from proxy service: " + wdaRealDeviceURL);
+
+
     Agent agent = agentService.findByUniqueId(agentUuid);
     AgentDevice agentDevice = agentDeviceService.findAgentDeviceByUniqueId(agent.getId(), deviceUuid);
+    String presignedUrl;
+    String filePath = storageServiceFactory.getStorageService().downloadFromRemoteUrl(wdaRealDeviceURL.toString());
     ProvisioningProfileDevice profileDevice = provisioningProfileDeviceService.findByAgentDeviceId(agentDevice.getId());
-    if (profileDevice != null) {
-      URL presignedUrl = storageServiceFactory.getStorageService().generatePreSignedURL("wda/"
-        + profileDevice.getProvisioningProfileId() + "/wda.ipa", StorageAccessLevel.READ, 180);
-      iosWdaResponseDTO.setWdaPresignedUrl(presignedUrl.toString());
-      log.info("Ios Wda Response DTO - " + iosWdaResponseDTO);
-      return iosWdaResponseDTO;
-    } else {
+    storageServiceFactory.getStorageService().addFile("wda/"
+            + profileDevice.getProvisioningProfileId() + "/wda.ipa", new File(filePath));
+    resignWdaService.reSignWda(profileDevice.getProvisioningProfile());
```

Comment:
@shabarish-testsigma Why do we need to sign the wda here ? Isn't it signed during provisioning profile creation ?

Reply:
@tarun-testsigma no wda is not signed other ipas are signed , wda is downloaded and stored only on execution(from a public url), thats the reason we are signing it here

Reply:
Have a look at ReSignTask.java(Line: 49). Here we are resigning wda for a provisioning profile

Reply:
@tarun-testsigma yes we should remove that as its redundant , will do it in the next build

Reply:
Please remove it now. Otherwise, resigning wda will keep failing whenever we upload a provisioning profile. Its a small change. Also, move this resigning line to wda service (installWdaToDevice)